src/controls/ObjectArrayControl.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+  Helpers,
+  ArrayLayoutProps,
+  composePaths,
+  createDefaultValue,
+  findUISchema,
+} from "@jsonforms/core"
+import {
+  JsonFormsDispatch,
+  withJsonFormsArrayLayoutProps,
+} from "@jsonforms/react"
+import { Flex, Form, List, Button } from "antd"
+import type { Rule } from "antd/es/form"
+import range from "lodash.range"
+import { useEffect, useMemo } from "react"
+import { ArrayControlOptions } from "../ui-schema"
+import { usePreviousValue } from "../common/usePrevious"
+import React from "react"
+
+export function ObjectArrayControl({
+  data,
+  enabled,
+  path,
+  schema,
+  uischema,
+  addItem,
+  removeItems,
+  renderers,
+  cells,
+  visible,
+  rootSchema,
+  uischemas,
+  required,
+}: ArrayLayoutProps) {
+  const foundUISchema = useMemo(() => {
+    return findUISchema(
+      uischemas ?? [],
+      schema,
+      uischema.scope,
+      path,
+      undefined,
+      uischema,
+      rootSchema,
+    )
+  }, [uischemas, schema, path, uischema, rootSchema])
+
+  const dataSource = useMemo(() => range(data), [data])
+
+  const addItemToList = useMemo(
+    () => addItem(path, createDefaultValue(schema, rootSchema)),
+    [addItem, path, rootSchema, schema],
+  )
+
+  const prevDataValue = usePreviousValue(data)
+  useEffect(() => {
+    if (data === 0 && prevDataValue === null) {
+      addItemToList()
+    }
+  })
```

Comment:
You'd normally include the dep array for the use effect, otherwise it almost serves no purpose as it runs in every re-render.

Reply:
Happy to debug this synchronously, but we were unable to get this to work any other way.